### PR TITLE
ToricVarieties: More improvements

### LINKED ISSUE #1280: ToricVarieties: Use IntegerUnion for code improvement

### PR DESCRIPTION
**Pull request by HereAround**
- Cohomology classes defined by (equ. class of) polynomial.
- Fix coefficient ring to `Q`. (Method which accept ring as first argument, and thereby allow different coefficient rings/variable names are and will be supported in the future.)
- Type stable return value for basis of line bundle sections.
- Use `IntegerUnion` to simplify the code.

This PR contains the changes in https://github.com/oscar-system/Oscar.jl/pull/1279, which should be merged first. Once this is done, the changes in this very PR should close https://github.com/oscar-system/Oscar.jl/issues/1280 and https://github.com/oscar-system/Oscar.jl/issues/1257.

### CODE REVIEW

File: src/ToricVarieties/NormalToricVarieties/constructors.jl [hunk 1]
```diff
@@ -181,9 +181,13 @@ A normal toric variety
...
-    fan = normal_fan(P)
-    variety = NormalToricVariety(fan)
+    #construct the variety
+    variety = NormalToricVariety(normal_fan(P))
+    
+    # set attributes
     set_attribute!(variety, :polyhedron, P)
+    
+    # return
     return variety
```
Comment (lkastner) [inline src/ToricVarieties/NormalToricVarieties/constructors.jl]:
```suggestion
    variety = NormalToricVariety(normal_fan(P))
    set_attribute!(variety, :polyhedron, P)
    return variety
```
To me it is a bit unclear why the comments are needed. E.g. the comment `# return` when the next line literally starts with `return`. Same for `# set attributes` when the next line literally calls `set_attribute!`. I suggest leaving out comments as much as possible as these are not tested and once the code is changed it often happens that old comments are left and leave to confusion. At least this is what I took from the chapter on comments in the clean code book.

**Review by lkastner — APPROVED** [review]

File: src/ToricVarieties/NormalToricVarieties/constructors.jl (hunk 1 above)
Comment (fingolfin) [reply src/ToricVarieties/NormalToricVarieties/constructors.jl]: I'd phrase it a bit different: comments should not state what you are doing (to see that, one can read the code after all -- or if the code is so opaque that one needs explanations, one should perhaps revise). But definitely have comments that explain *why* you are doing certain things; comments should help illuminate the bigger picture. But of course that's just rule of thumb, too.

In any case, I agree these comments here are redundant

Comment (HereAround) [reply src/ToricVarieties/NormalToricVarieties/constructors.jl]: You are both right. I will adjust this (hopefully very soon).

**Review by fingolfin — APPROVED** [review]
Looks good to me

**Pull request merged**